drivers/docker/handle.go

```diff
@@ -125,12 +125,24 @@ func (h *taskHandle) Kill(killTimeout time.Duration, signal os.Signal) error {
 	// Only send signal if killTimeout is set, otherwise stop container
 	if killTimeout > 0 {
 		if err := h.Signal(signal); err != nil {
-			return err
-		}
-		select {
-		case <-h.waitCh:
-			return nil
-		case <-time.After(killTimeout):
+			// Container has already been removed.
+			if strings.Contains(err.Error(), NoSuchContainerError) {
+				h.logger.Debug("attempted to signal nonexistent container")
+				return nil
+			}
+			// Container has already been stopped.
+			if strings.Contains(err.Error(), ContainerNotRunningError) {
+				h.logger.Debug("attempted to signal a not-running container")
+				return nil
+			}
+
+			h.logger.Error("failed to signal container while killing", "error", err)
```

Comment:
Do we want to fully squelch the error here? I could see an error where the docker daemon stopped or some other transport issue occurred and we'd want to retry.

Reply:
There are many conditionals, so maybe a comment that if signal failed, we'd still attempt to stop it again in this case with `StopContainer` call below.  This feels like an accidental drop of `return err`.

Also, the detection for stopped containers can be extracted to a function.

Reply:
@notnoop Added a comment. I didn't move it to a function because the logging is different and a little duplication for clarity felt fine to me.